src/main/java/com/thoughtworks/aceleradora/servicos/CategoriaServico.java

```diff
@@ -15,8 +19,32 @@ public CategoriaServico(CategoriaRepositorio repositorio) {
         this.repositorio = repositorio;
     }
 
-    public List<Categoria> pegarCategorias() {
-        return repositorio.findAll();
+    public Resposta<List<Categoria>> pegarCategorias() {
+
+        try {
+            return new Resposta<List<Categoria>>(null, repositorio.findAll());
```

Comment:
Resposta poderia ter um construtor que recebe apenas dados:
```java
Resposta(T dados) {
  this.mensagem = "";
  this.dados = dados;
}
```

Aqui ficaria assim:

```java
return new Resposta<>(repositorio.findAll());
```

Essa classe poderia inclusive ter somente construtores estáticos (como `Optional.of`, por exemplo), assim poderíamos nomeá-los e teríamos mais controle sobre os estados de um objeto Resposta:

```java
private Resposta(String mensagem, T dados) {
  // sets
}

public static <R> Resposta<R> semMensagens(R dados) {
  return new Resposta<>("", dados);
}

public static <R> Resposta<R> comMensagem(String mensagem, R dados) {
  return new Resposta<>(mensagem, dados);
}
```